src/posix/platform/hdlc_interface.hpp

```diff
@@ -36,17 +36,18 @@
 
 #include "platform-config.h"
 #include "spinel_interface.hpp"
-
 #include "ncp/hdlc.hpp"
 
+#if OPENTHREAD_POSIX_RCP_UART_ENABLE
+
 namespace ot {
 namespace PosixApp {
 
 /**
  * This class defines an HDLC interface to the Radio Co-processor (RCP)
  *
  */
-class HdlcInterface : public SpinelInterface
+class HdlcInterface
```

Comment:
use inheritance?

Reply:
We don't need to use inheritance. If you are looking at the content of the class `SpinelInterface`, no items should be inherited.